detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/IgnoredReturnValue.kt

```diff
@@ -49,8 +54,14 @@ class IgnoredReturnValue(config: Config = Config.empty) : Rule(config) {
     )
 
     @Configuration("if the rule should check only annotated methods")
+    @Deprecated("Use `restrictToConfig` instead")
     private val restrictToAnnotatedMethods: Boolean by config(defaultValue = true)
 
+    @Suppress("DEPRECATION")
+    @OptIn(UnstableApi::class)
+    @Configuration("If the rule should check only methods matching to configuration, or all methods")
+    private val restrictToConfig: Boolean by configWithFallback(::restrictToAnnotatedMethods, defaultValue = true)
```

Comment:
Naming is so difficult. I think your naming is good. I'm just proposing other but I'm not sure if it's better. What do you think?

```suggestion
    @Suppress("DEPRECATION")
    @OptIn(UnstableApi::class)
    @Configuration("If the rule should check all methods ignoring the annotations or return types.")
    private val checkAllMethods: Boolean by configWithFallback({ !restrictToAnnotatedMethods }, defaultValue = true)
```

Reply:
Yes that's better

Reply:
But when I try to do it, I get the error:
<img width="474" alt="image" src="https://user-images.githubusercontent.com/3684355/183012980-1655394b-dfae-4bc2-9733-4f70dc1ff008.png">

The options I see:
1. Add the new parameter to `configWithFallback` that will transform fallback value
2. Keep the new property equal to `restrictToAnnotatedMethods`